source/postprocess/visualization/boundary_velocity_residual.cc

```diff
@@ -0,0 +1,260 @@
+/*
+  Copyright (C) 2011 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+#include <aspect/simulator.h>
+#include <aspect/postprocess/visualization/boundary_velocity_residual.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      BoundaryVelocityResidual<dim>::
+	  BoundaryVelocityResidual ()
+        :
+        DataPostprocessorVector<dim> ("boundary_velocity_residual",
+        			update_values | update_quadrature_points | update_gradients)
+      {}
+
+
+      template <int dim>
+      void
+	  BoundaryVelocityResidual<dim>::initialize ()
+	  {
+      // The input gpaltes file with default values of pointone and pointtwo
+	    pointone[0] = 1.570796; pointone[1] = 0;
+	    pointtwo[0] = 1.570796; pointtwo[1] = 1.570796;
+
+	   gplates_lookup =  std_cxx14::make_unique<BoundaryVelocity::internal::GPlatesLookup<dim> > (
+			   pointone, pointtwo);
+
+	   gplates_lookup->load_file(gplates_data_directory + gplates_data_file_name, this->get_mpi_communicator());
+
+	 // The input ascii table contains one data column (velocity components) in addition to the coordinate columns.
+	   ascii_data_lookup = std_cxx14::make_unique<Utilities::AsciiDataLookup<dim> >(dim, ascii_scale_factor);
+	   ascii_data_lookup->load_file(ascii_data_directory + ascii_data_file_name, this->get_mpi_communicator());
```

Comment:
could you decide this here as:

```suggestion
if (use_ascii_data == false)
{
	   gplates_lookup =  std_cxx14::make_unique<BoundaryVelocity::internal::GPlatesLookup<dim> > (
			   pointone, pointtwo);

	   gplates_lookup->load_file(gplates_data_directory + gplates_data_file_name, this->get_mpi_communicator());
}
else
{
	 // The input ascii table contains one data column (velocity components) in addition to the coordinate columns.
	   ascii_data_lookup = std_cxx14::make_unique<Utilities::AsciiDataLookup<dim> >(dim, ascii_scale_factor);
	   ascii_data_lookup->load_file(ascii_data_directory + ascii_data_file_name, this->get_mpi_communicator());
	   }
```